gravitee-apim-gateway/gravitee-apim-gateway-http/src/main/java/io/gravitee/gateway/jupiter/http/vertx/VertxHttpServerRequest.java

```diff
@@ -89,51 +83,62 @@ public boolean isWebSocketUpgraded() {
 
     @Override
     public Maybe<Buffer> body() {
-        return bodyChunksFlowable.body();
+        return bufferFlow.body();
     }
 
     @Override
     public Single<Buffer> bodyOrEmpty() {
-        return bodyChunksFlowable.bodyOrEmpty();
+        return bufferFlow.bodyOrEmpty();
     }
 
     @Override
     public void body(final Buffer buffer) {
-        bodyChunksFlowable.body(buffer);
+        bufferFlow.body(buffer);
     }
 
     @Override
     public Completable onBody(final MaybeTransformer<Buffer, Buffer> onBody) {
-        return bodyChunksFlowable.onBody(onBody);
+        return bufferFlow.onBody(onBody);
     }
 
     @Override
     public Flowable<Buffer> chunks() {
-        return bodyChunksFlowable.chunks();
+        return bufferFlow.chunks();
     }
 
     @Override
     public void chunks(final Flowable<Buffer> chunks) {
-        bodyChunksFlowable.chunks(chunks);
+        bufferFlow.chunks(chunks);
     }
 
     @Override
     public Completable onChunks(final FlowableTransformer<Buffer, Buffer> onChunks) {
-        return bodyChunksFlowable.onChunks(onChunks);
+        return bufferFlow.onChunks(onChunks);
     }
 
     @Override
     public void messages(final Flowable<Message> messages) {
-        httpMessages.messages(messages);
+        getMessageFlow().messages(messages);
+
+        // If message flow is set up, make sure any access to chunk buffers will not be possible anymore and returns empty.
+        chunks(Flowable.empty());
```

Comment:
Should we use a lock boolean instead and so test it while accessing any method related to chunks throwing IllegalState or something in the case we are accessing body on sme request ? BEcause settings empty while just make debug difficult in those cases.

Reply:
Humm.... honestly I'm not super sure of the behavior to adopt for now. I suggest to start working on the policy execution to get more insight on what is preferable to do.
Ok for you ?

Reply:
Yes that' s fine to me.